io/io/v7/inc/ROOT/TFile.hxx

```diff
@@ -49,8 +49,7 @@ private:
 
   /// Serialize the object at address, using the object's TClass.
   //FIXME: what about `cl` "pointing" to a base class?
-  void WriteMemoryWithType(std::string_view name, const void *address,
-                           TClass *cl);
+  void WriteMemoryWithType(const std::string &name, const void *address, TClass *cl);
```

Comment:
If the interface uses the set of characters as null terminated string (by definition, not just because the current implementation happens to do that) then a `std::string` is fine. If we do string manipulation anyway (extract a path name, remove `"?CACHE=3M&CERT=..."` from a URL) then there's no point in forcing the parameter to be copied into a null-terminated `std::string`.